agent/exec/dockerapi/executor.go

```diff
@@ -17,15 +17,23 @@ type executor struct {
 	client           engineapi.APIClient
 	secrets          exec.SecretsManager
 	genericResources []*api.GenericResource
+	node             *api.NodeDescription
 }
 
 // NewExecutor returns an executor from the docker client.
 func NewExecutor(client engineapi.APIClient, genericResources []*api.GenericResource) exec.Executor {
-	return &executor{
+	var executor = &executor{
 		client:           client,
 		secrets:          secrets.NewManager(),
 		genericResources: genericResources,
 	}
+	// Get the node information from the newly created executor
+	var ctx = context.Background()
```

Comment:
Using `context.Background` here is problematic. It meanst `NewExecutor` can potentially block forever.

I would make `Describe` set `executor.node`. `Describe` is called when the agent first starts up, before anything else happens. It is also called periodically to refresh the information.

We should protect `executor.node` with a mutex so it isn't accessed during an update.